### PR TITLE
Feature: complete Registry pallet implementation

### PR DESCRIPTION
**Pull request by FaisalAl-Tameemi**
This `registry` pallet mainly resolves the need for having a list of API feeds registered with the chain.

It mainly exposes 2 methods:

1. Registering
2. Unregistering

Each feed must contain relevant information such as:
1. Block number registration occurred at
2. The URL of the feed
3. The status of the feed (useful when introducing slashing later on, defaults to "Registered")
4. The path of the data within the feed responses (this will potentially be passed into the corresponding phat contract)

### CODE REVIEW

File: pallets/registry/src/lib.rs [hunk 1]
```diff
@@ -25,70 +25,93 @@ pub mod pallet {
...
+    pub enum ApiFeedStatus {
+        Registered,
+        Completed,
+    }
+
+    impl Default for ApiFeedStatus {
+        fn default() -> Self {
+            ApiFeedStatus::Registered
+        }
+    }
```
Comment (FaisalAl-Tameemi) [inline pallets/registry/src/lib.rs:44]: This should be moved into a separate `types.rs` file within the same pallet.

File: pallets/registry/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+use frame_support::pallet_prelude::{Decode, Encode, MaxEncodedLen, TypeInfo};
+use frame_support::{
+    traits::{ConstU128, ConstU32, ConstU64},
+    BoundedVec,
+};
+use crate::Config;
+
+/// Types representing limited strings
+pub type RegistryFeedKey<T> = BoundedVec<u8, <T as Config>::MaxKeySize>;
+pub type RegistryFeedUrl<T> = BoundedVec<u8, <T as Config>::MaxStringSize>;
```
Comment (a-moreira) [inline pallets/registry/src/types.rs:10]:
```suggestion
pub type RegistryFeedUrl<T> = BoundedVec<u8, <T as Config>::MaxUrlSize>;
```
just for consistency, wdyt?

File: pallets/registry/src/lib.rs [hunk 3]
```diff
@@ -97,54 +96,83 @@ pub mod pallet {
...
-            if feed_exists {
-                let feed = Self::api_feeds(&who, &key).unwrap();
+            // Remove the feed from storage if it exists.
+            // This also implicitly checks that the feed is owned by the origin account.
+            if let Some(feed) = <ApiFeeds<T>>::get(&who, &key) {
                 <ApiFeeds<T>>::remove(&who, &key);
-                Self::deposit_event(Event::UnRegFeed {
-                    sender: who,
+                Self::deposit_event(Event::FeedUnregistered {
+                    by: who,
```
Comment (a-moreira) [inline pallets/registry/src/lib.rs:169]:
```suggestion
                    caller: who,
```
I think `caller`  would be nice (in the contract of substrate)

**Review by a-moreira — APPROVED** [review]
just some nits

File: pallets/registry/src/types.rs (hunk 2 above)
Comment (FaisalAl-Tameemi) [reply pallets/registry/src/types.rs:10]: Agreed

**Pull request merged**